uspace/lib/draw/font/bitmap_backend.c

```diff
@@ -211,6 +211,7 @@ static void bb_release(void *backend_data)
 	free(data->glyph_cache);
 
 	data->decoder->release(data->decoder_data);
+	free(data);
```

Comment:
Not sure if this is a good location for freeing backend_data for symmetry reasons. font_create() is already passed allocated backend_data, so shouldn't the free go after the call to release in font_release()?

Reply:
I am not sure, I mean, font_release() does the following:

``font->backend->release(font->backend_data);``

Isn't the backend's release() callback supposed to fully free the backend's data?

Reply:
That's the question. I'd say yes if the backend allocated the data, but that does not seem to be the  case. Or at least the ownership should be better documented. Can @martin-sucha help?

Reply:
The backend_data is allocated in bitmap_font_create() (bitmap_backend.c line 231), so the ownership belongs to the backend.

I think we should consider adding the a pointer to backend_data directly into the font_backend structure.